### DISCUSSION TITLE
how to change comment sign by default

### DISCUSSION THREAD

**Discussion by whoizit** [helix-editor/helix#5096] [Q&A]
For my workflow, comment by default is `#`, not `//`. But I have always `//`

**Comment by the-mikedavis** [comment]
This can be configured in the [`languages.toml`](https://docs.helix-editor.com/master/languages.html). See all of the `comment-token` attributes in https://github.com/helix-editor/helix/blob/master/languages.toml

**Comment by whoizit** [comment]
this not works for me:
```
[[language]]                                                                                                                                            
name = "default"                                                                                                                                        
scope = "scope.default"                                                                                                                                 
roots = []                                                                                                                                              
file-types = []                                                                                                                                                                                                                                                                     
comment-token = "#"
```
```
Bad language config: invalid type: map, expected a string for key `language.file-types`
```

**Reply by the-mikedavis** [reply]
It isn't currently possible to configure these settings for all languages - is that what you're looking for? It must be done per-language currently.

For the error: are you running on 22.08.1 and did you copy the `languages.toml` from the repository? The `languages.toml` on master is not backwards compatible with 22.08.1 and emits this error.

**Reply by whoizit** [reply]
I want open any configuration file in Linux and use # instead //

**Reply by ModProg** [reply]
> It isn't currently possible to configure these settings for all languages - is that what you're looking for? It must be done per-language currently.

It would be great if we could configure the fallback used when no language is detected. (an issue I raised in #9165 which was closed in favor of an eventual improvement in #8853).

For now, if anyone finds this, my solution is to convert all the file types to globs with 
```sh
curl https://raw.githubusercontent.com/helix-editor/helix/master/languages.toml | tomlq "{language: .language | map({name, scope, \"file-types\": .[\"file-types\"] | map({glob}? // {\"glob\": \"*.\\(.)\"})})}" -t >> ~/.config/helix/languages.toml
```

this allows me to define a fallback language to allow me to configure it
```toml
[[language]]
name = "fallback"
scope = ""
file-types = [{glob = "*"}]
comment-tokens = ["#"]
```

The conversion to `glob` seems to be necessary because globs take precedence over normal file extensions. Also make sure to move the `fallback` language to the top of the list, so the other globs take precedence over the fallback. (Not sure about the performance implications)

**Comment by pandalec** [comment]
```
[[language]]
name = "conf"
comment-token = "#"
scope = ""
file-types = ["conf"]
roots = []
```

quite old I know but I searched for a solution and stumbled open this thread. Just adding my current config if anyone else searches
